apps/www/components/Audio/AudioPlayerContainer.tsx

```diff
@@ -0,0 +1,622 @@
+        return
+      }
+      const nextUp = audioQueue[0]
+      setActivePlayerItem(nextUp)
+      setIsVisible(true)
+      await onPlay()
+    } catch (error) {
+      handleError(error)
+    }
+  }
+
+  const onError = () => {
+    if (mediaRef.current && mediaRef.current.error) {
+      const error = mediaRef.current.error
+      const errorObject = {
+        message: error.message,
+        code: error.code,
+        MEDIA_ERR_ABORTED: error.MEDIA_ERR_ABORTED,
+        MEDIA_ERR_NETWORK: error.MEDIA_ERR_NETWORK,
+        MEDIA_ERR_DECODE: error.MEDIA_ERR_DECODE,
+      }
+      trackEvent([
+        AUDIO_PLAYER_TRACK_CATEGORY,
+        'webPlayer',
+        'error',
+        JSON.stringify(errorObject),
+      ])
+      handleError(new Error(JSON.stringify(errorObject, null, 2)))
+      setHasError(true)
+    }
+  }
+
+  useInterval(
+    saveActiveItemProgress,
+    isPlaying ? SAVE_MEDIA_PROGRESS_INTERVAL : null,
+  )
+
+  // Reset media-element if new source is provided
+  useEffect(() => {
+    if (activePlayerItem?.id === trackedPlayerItem?.current?.id) {
+      return
+    }
+    if (
+      mediaRef.current &&
+      // If no data could be retrieved so far, manually trigger load
+      mediaRef.current.readyState === 0
+    ) {
+      setIsLoading(true)
+      mediaRef.current.load()
+      setIsLoading(false)
+    }
+  }, [activePlayerItem, trackedPlayerItem, setIsLoading, setHasAutoPlayed])
+
+  // Sync web-ui with web media-element
+  useEffect(() => {
+    // Update the internal state based on the audio element every 500ms
+    if (isPlaying && mediaRef.current) {
+      const interval = setInterval(() => {
+        syncWithMediaElement()
+      }, Math.min(DEFAULT_SYNC_INTERVAL / playbackRate, 1000))
+      return () => clearInterval(interval)
+    }
+  }, [syncWithMediaElement, isPlaying, playbackRate])
+
+  // Web
+  const handleItemPushedToFront = useCallback(() => {
+    const nextActivePlayerItem = audioQueue[0]
+    const alreadyHadActivePlayerItem = !!activePlayerItem
+    setActivePlayerItem(nextActivePlayerItem)
+    setShouldAutoPlay(isPlaying)
+    setHasAutoPlayed(false)
+    setIsPlaying(false)
+
+    // Provide the inital duration and progress for the ui
+    // as a fallback until the media was loaded
+    // to prevent the ui from showing 0:00
+
+    const audioSource = nextActivePlayerItem.document?.meta?.audioSource
+    if (audioSource) {
+      setDuration(audioSource.durationMs / 1000)
+      if (audioSource?.userProgress) {
+        setCurrentTime(audioSource.userProgress.secs)
+      }
+
+      if (mediaRef.current && alreadyHadActivePlayerItem) {
+        mediaRef.current.load()
+      }
+    }
+  }, [audioQueue, activePlayerItem, trackedPlayerItem, isPlaying, mediaRef])
+
+  if (
+    isClient &&
+    audioQueue &&
+    audioQueue.length > 0 &&
+    (!activePlayerItem ||
+      activePlayerItem?.document?.id !== audioQueue[0].document.id)
+  ) {
+    handleItemPushedToFront()
+  }
```

Comment:
Tripple-check if this is the right place and way (instead of using useEffect) to do this